float32 velocity
float32 time
float32 radius